lib/spack/spack/schema/packages.py

```diff
@@ -72,17 +67,39 @@
                                 'type': 'array',
                                 'default': [],
                                 'items': {'type': 'string'}, }, }, },
-                    'paths': {
-                        'type': 'object',
-                        'default': {},
-                    },
                     'variants': {
                         'oneOf': [
                             {'type': 'string'},
                             {'type': 'array',
                              'items': {'type': 'string'}}],
                     },
+                    'externals': {
+                        'type': 'array',
+                        'items': {
+                            'type': 'object',
+                            'properties': {
+                                'spec': {'type': 'string'},
+                                'prefix': {'type': 'string'},
+                                'modules': {'type': 'array',
+                                            'items': {'type': 'string'}},
+                                'extra_attributes': {'type': 'object'}
+                            },
+                            'additionalProperties': True,
```

Comment:
Does this need `additionalProperties: True` now that there is an explicit `extra_attributes` dictionary?

Reply:
I believe for the time being it's a better approach. This has the drawback of not signalling typos in the configuration file, but permits us to add keywords at this level in the future without causing a hard forward-incompatibility. It will just happen that if we add extra attributes at this level a "old" version of Spack will not make use of them but will nonetheless be able to use the rest of `packages.yaml`.